### PR TITLE
updating to windows 0.58

### PR DESCRIPTION
**Pull request by psandana**
- Updating dependency to consume `windows@0.58.0`
- Adapting code to new API
- Adapting test to new API
- Bumping version `v0.2.0`

### CODE REVIEW

**Comment by psandana** [comment]
Hi @vsrs. I found this project, and it is very interesting. I need something like this for `windows` crate version `0.58.0`. Would you like my contribution to upgrade this package to be compatible with it?

If you accept the changes, it would be required that a new version is released. I already did changes to update the version to `v0.2.0`.

Thanks,
Pato!

File: tests/bcrypt_hash.rs [hunk 1]
```diff
@@ -14,22 +14,30 @@ impl AlgProvider {
...
                 BCRYPT_OPEN_ALGORITHM_PROVIDER_FLAGS::default(),
-            )?
+            )
         };
 
+        if result.is_err() {
+            return Err(result.into());
+        }
+
         Ok(handle)
```
Comment (vsrs) [inline tests/bcrypt_hash.rs]:
```rust
result.to_hresult().map(|| handle)
```

File: tests/bcrypt_hash.rs [hunk 2]
```diff
@@ -14,22 +14,30 @@ impl AlgProvider {
...
-            BCryptCreateHash(self, handle.as_mut(), None, None, 0)?;
+        let result = unsafe {
+            BCryptCreateHash(self.0, handle.as_mut(), None, None, 0)
+        };
+
+        if result.is_err() {
+            return Err(result.into());
         }
 
         Ok(handle)
```
Comment (vsrs) [inline tests/bcrypt_hash.rs]:
```
result.to_hresult().map(|| handle)
```

File: tests/bcrypt_hash.rs [hunk 3]
```diff
@@ -40,26 +48,32 @@ safe_handle!(pub Hash(BCRYPT_HASH_HANDLE as BCRYPT_HANDLE), BCryptDestroyHash);
...
-            )?;
+            )
+        };
 
-            Ok(length)
+        if result.is_err() {
+            return Err(result.into());
         }
+
+        Ok(length)
```
Comment (vsrs) [inline tests/bcrypt_hash.rs]:
```
result.to_hresult().map(|| length)
```

**Review by vsrs — CHANGES_REQUESTED** [review]
@psandana, thank you for your contribution :)

LGTM! I have only minor suggestions.

Also, it might be worth mentioning in the README that v1 can be used for older windows crate versions.

**Comment by vsrs** [comment]
@psandana,
I've published the new version. Thank you once again.

**Pull request merged**